romancal/regtest/test_wfi_pipeline.py

```diff
@@ -95,6 +98,42 @@ def test_level2_image_processing_pipeline(rtdata, ignore_asdf_paths):
                       passfail(model.meta.cal_step.saturation == 'COMPLETE'))
     assert model.meta.cal_step.saturation == 'COMPLETE'
 
+    # DMS-129 tests for WFI mode
+    if model.meta.exposure.type == 'WFI_IMAGE':
+        # check if assign_wcs step is complete
+        pipeline.log.info('DMS-129 MSG: Status of the step:             assign_wcs    ' +
+                      str(model.meta.cal_step.assign_wcs))
+        pipeline.log.info('DMS-129 MSG: Testing completion of WCS assignment in'
+                        'Level 2 image output.......' +
+                        passfail(model.meta.cal_step.assign_wcs == 'COMPLETE'))
+        assert model.meta.cal_step.assign_wcs == 'COMPLETE'
+        # check if WCS exists
+        pipeline.log.info('DMS-129 MSG: Testing that a WCS object exists    ')
+        pipeline.log.info('DMS-129 MSG: Testing that WCS exists in'
+                        'Level 2 image output.......' +
+                        passfail(model.meta.wcs is not None))
+        assert model.meta.wcs is not None
+        pipeline.log.info('DMS-129 MSG: Checking that transformation from detector to telescope coords has been applied in'
+                        'Level 2 image output.......' +
+                        passfail('v2v3' in model.meta.wcs.available_frames))
+        assert 'v2v3' in model.meta.wcs.available_frames
+        # compare coordinates before and after distortion correction has been applied
+        # 1 - get new image array based on the model
+        x0, y0 = grid_from_bounding_box(model.meta.wcs.bounding_box)
+        # 2 - apply the distortion-corrected WCS solution to new image array
+        corrected_coords = model.meta.wcs(x0, y0)
+        # 3 - apply the transformation from 'v2v3' to 'world' without distortion correction
+        original_coords = model.meta.wcs.get_transform('v2v3', 'world')(x0, y0)
+        # compare both results to make sure they don't match
+        # (which means the distortion correction was actually applied to the model)
+        assert (corrected_coords[0] != original_coords[0]).all()
+        assert (corrected_coords[1] != original_coords[1]).all()
+        
+        pipeline.log.info('DMS-129 MSG: Checking that distortion correction has been applied in'
+                        'Level 2 image output.......' +
+                        passfail('v2v3' in model.meta.wcs.available_frames))
+        assert 'v2v3' in model.meta.wcs.available_frames
```

Comment:
This is already on line 19 - remove?

Reply:
That duplicated piece of code has been removed.